pkg/vpcmodel/diffSubnets_test.go

```diff
@@ -0,0 +1,90 @@
+		&mockSubnet{"10.1.20.0/22", "subnet1", []Node{cfg1.Nodes[0]}},
+		&mockSubnet{"10.2.20.0/22", "subnet2", []Node{cfg1.Nodes[1]}})
+	cfg1.NodeSets = append(cfg1.NodeSets, &mockSubnet{"10.3.20.0/22", "subnet3", []Node{cfg1.Nodes[2]}})
+
+	cfg2 := &CloudConfig{Nodes: []Node{}, NodeSets: []NodeSet{}}
+	cfg2.Nodes = append(cfg2.Nodes,
+		&mockNetIntf{cidr: "10.3.20.5/32", name: "vsi2-1"},
+		&mockNetIntf{cidr: "10.7.20.6/32", name: "vsi2-2"},
+		&mockNetIntf{cidr: "10.9.20.7/32", name: "vsi2-3"})
+	cfg2.NodeSets = append(cfg2.NodeSets, &mockSubnet{"10.2.20.0/22", "subnet2", []Node{cfg2.Nodes[0]}},
+		&mockSubnet{"10.3.20.0/22", "subnet3", []Node{cfg2.Nodes[1]}},
+		&mockSubnet{"10.4.20.0/22", "subnet4", []Node{cfg2.Nodes[2]}})
+
+	subnetConnMap1 := &VPCsubnetConnectivity{AllowedConnsCombined: NewSubnetConnectivityMap()}
+	subnetConnMap1.AllowedConnsCombined.updateAllowedSubnetConnsMap(cfg1.NodeSets[0], cfg1.NodeSets[1], common.NewConnectionSet(true))
+	subnetConnMap1.AllowedConnsCombined.updateAllowedSubnetConnsMap(cfg1.NodeSets[1], cfg1.NodeSets[2], common.NewConnectionSet(true))
+	subnetConnMap1.AllowedConnsCombined.updateAllowedSubnetConnsMap(cfg1.NodeSets[3], cfg1.NodeSets[1], common.NewConnectionSet(true))
+	subnetConnMap1.AllowedConnsCombined.updateAllowedSubnetConnsMap(cfg1.NodeSets[2], cfg1.NodeSets[3], common.NewConnectionSet(true))
+	subnetConnMap1.AllowedConnsCombined.updateAllowedSubnetConnsMap(cfg1.NodeSets[3], cfg1.NodeSets[2], common.NewConnectionSet(true))
+
+	subnetConnMap2 := &VPCsubnetConnectivity{AllowedConnsCombined: NewSubnetConnectivityMap()}
+	subnetConnMap2.AllowedConnsCombined.updateAllowedSubnetConnsMap(cfg2.NodeSets[1], cfg2.NodeSets[0], common.NewConnectionSet(true))
+	subnetConnMap2.AllowedConnsCombined.updateAllowedSubnetConnsMap(cfg2.NodeSets[1], cfg2.NodeSets[2], common.NewConnectionSet(true))
+
+	subnetConfigConn1 = &SubnetConfigConnectivity{cfg1, subnetConnMap1.AllowedConnsCombined}
+	subnetConfigConn2 = &SubnetConfigConnectivity{cfg2, subnetConnMap2.AllowedConnsCombined}
+
+	return subnetConfigConn1, subnetConfigConn2
+}
+
+func TestSimpleSubnetSubtract(t *testing.T) {
+	subnetConfigConn1, subnetConfigConn2 := configSimpleSubnetSubtract()
+	subnet1Subtract2 := subnetConfigConn1.SubnetConnectivitySubtract(subnetConfigConn2)
+	subnet1Subtract2Str := subnet1Subtract2.EnhancedString(true)
+	fmt.Printf("subnet1Subtract2:\n%v\n", subnet1Subtract2Str)
+	newLines := strings.Count(subnet1Subtract2Str, "\n")
+	// there should be 4 lines in subnet1Subtract2Str
+	require.Equal(t, 4, newLines)
+	require.Contains(t, subnet1Subtract2Str, "-- subnet3 => subnet1 :  missing destination")
+	require.Contains(t, subnet1Subtract2Str, "-- subnet2 => subnet3 :  missing connection")
+	require.Contains(t, subnet1Subtract2Str, "-- subnet0 => subnet1 :  missing source and destination")
+	require.Contains(t, subnet1Subtract2Str, "-- subnet1 => subnet2 :  missing source")
```

Comment:
Extend to include the case of a changed connection

Reply:
1. Currently the functionality used to compute the delta between connections is an old one
2. There is no example of unit test with non-trivial connection, and we lack the required functionality for that which needs to be added. So I suggest we defer this to the PR in which we will deal with connections (in the near future).

Reply:
You can create in the `VPCsubnetConnectivity` an edge between two subnets with a connection-set that is not the trivial `NewConnectionSet(true)`, such as in https://github.com/np-guard/vpc-network-config-analyzer/blob/main/pkg/ibmvpc/connectivityAnalysis_test.go#L95-L99